crates/miden-testing/src/kernel_tests/block/proven_block_success.rs

```diff
@@ -147,9 +153,15 @@ async fn proven_block_success() -> anyhow::Result<()> {
     // Prove block.
     // --------------------------------------------------------------------------------------------
 
-    let proven_block = LocalBlockProver::new(MIN_PROOF_SECURITY_LEVEL)
-        .prove_dummy(proposed_block)
-        .context("failed to prove proposed block")?;
+    let (header, body) = build_block(proposed_block.clone())?;
+    let inputs = chain.get_block_inputs(batches.iter()).unwrap();
+    let ordered_batches = OrderedBatches::new(batches);
+    let block_proof = LocalBlockProver::new(MIN_PROOF_SECURITY_LEVEL).prove_dummy(
+        ordered_batches,
+        header.clone(),
+        inputs,
+    )?;
+    let proven_block = ProvenBlock::new_unchecked(header, body, block_proof);
```

Comment:
As mentioned in another comment, I'd remove these code blocks (multiple here) with `chain.prove_block(proposed_block)?`.